### ISSUE TITLE
very nice, are you aware of yankpad?

### ISSUE THREAD

**Issue by zeltak** [tuhdo/org-recipes#2]
Hi 

Thx for this nice addition to emacs. I have been using something very similar i love called called called yankpad: https://github.com/Kungsgeten/yankpad

perhaps it would be interesting for you to borrow some ideas from yankpad or perhaps you both can collaborate and incorporate ideas since they seem very similar 
best

Z

**Comment by tuhdo** [comment]
> Thx for this nice addition to emacs. I

Glad to hear that.

When I was developing this tool, I was not aware of `yankpad`. Though similar, `yankpad` is still different from what I want. Originally, when developing this package, I want to quickly navigate the code snippets in my personal wiki (managed with `org-wiki`), and quickly collect and insert into the current buffer instead of manually extract blocks of code. Then, I want snippets to be chainable.

I will try to make this package complement with `yankpad`.

**Comment by Kungsgeten** [comment]
Hi Tuhdo! I'm the developer of `yankpad`. I haven't used `org-recipes`, but will check it out. I too think that it may be possible to merge the two packages (seems like similar use cases), but will have to see how the features differ.

**Comment by tuhdo** [comment]
@Kungsgeten Hi,

I'm happy to merge my repo into your repo, since it is already on MELPA and the features of this package can be useful to more people. 

My package can complement yours without overlapping. With your package, people collect simple snippets into org headers for quick expansion. With my package, people can write prose about their code, with little snippets in between text paragraphs; whenever you need a only the code, it can quickly collect all the little snippets and insert into the current buffer. 

For example, [this file](https://gist.github.com/tuhdo/e0d5d8a53389f03b438acc12bba4b853) is the org version of [Bit Twiddling Hacks](https://graphics.stanford.edu/~seander/bithacks.html). sing Helm, I can quickly access the document. For familiar tricks, using only a heading text, I know what the code is doing without even looking; instead, I simply insert all the code under a header with a simple `C-c i` key binding.

I will make a PR this weekend if you would like.

**Comment by Kungsgeten** [comment]
I see! It seems a bit like litterate programming combined with snippets? My guess would be that headlines (in `yankpad`) should have a special tag if they should be treated as "litterate programming". Another option could be to have a list of separate files (in `yankpad` there's only one file at the moment, containing all snippets) and that those are treated as litterate.

**Comment by tuhdo** [comment]
> I see! It seems a bit like litterate programming combined with snippets? My guess would be that headlines (in yankpad) should have a special tag if they should be treated as "litterate programming". Another option could be to have a list of separate files (in yankpad there's only one file at the moment, containing all snippets) and that those are treated as litterate.

Yes. Literate programming is already available as default in Org. This package only makes it easier to manage the code within the text, as listed in the feature list:

- List all Org headings with at least a code snippet.

- Each entry can be visited for viewing.

- Insert code from a recipes into the current buffer, or a specific file if the code block explicitly specifies. The description text is stripped and all code snippets are concatenated into a single snippet and then it is insert into current buffer.

- Compose recipes to generate code all at once: You can compose from the code under many headings, in which each heading can contain more than 1 snippet, into a bigger code snippet.

**Comment by Kungsgeten** [comment]
I don't know if you ever looked at the `yankpad` code. However I have now added a feature to `yankpad` which lets you write source blocks inside a snippet, and then concatenates the blocks upon snippet expansion. It is similar to `org-recipes` and I'm mentioning this package in the `yankpad` README.
